# https://jupyterbook.org/customize/config.html

title: Computing in Context (SIPA)
author: Aidan Feldman

# https://jupyterbook.org/en/stable/structure/configure.html#disable-building-files-that-arent-in-the-table-of-contents
only_build_toc_files: true
exclude_patterns:
  - .github/*
  - .pytest_cache/*
  - .venv/*

repository:
  url: https://github.com/afeld/computing-in-context
  path_to_book: site/

execute:
  execute_notebooks: "off"

html:
  # useful since the site's being deployed to both ReadTheDocs and GitHub Pages
  baseurl: https://computing-in-context.afeld.me/
  # https://jupyterbook.org/customize/config.html#add-a-link-to-your-repository
  use_repository_button: true
  extra_footer: |
    <div>
      <a href="https://creativecommons.org/publicdomain/zero/1.0/">
        <img src="https://mirrors.creativecommons.org/presskit/buttons/88x31/svg/cc-zero.svg" alt="Creative Commons 0 1.0" />
      </a>
    </div>

sphinx:
  config:
    # https://jupyterbook.org/en/stable/interactive/interactive.html#plotly
    html_js_files:
      - https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.7/require.min.js
    html_show_copyright: false
    # https://jupyterbook.org/advanced/html.html#configuring-to-improve-accessibility
    language: en
    linkcheck_allow_unauthorized: true
    # https://www.sphinx-doc.org/en/master/usage/configuration.html#options-for-the-linkcheck-builder
    linkcheck_anchors_ignore:
      # GitHub no longer has #readme anchors, apparently
      - "^readme$"
    linkcheck_ignore:
      # their anchors are title-cased
      - ^https://plotly.com
      # missing anchor
      # https://github.com/mattcone/markdown-guide/pull/214
      - https://www.markdownguide.org/basic-syntax/#headings
      # invalid cert
      - ^https://www.open311.org
      # behind Cloudflare; get 403s
      - ^https://(.+\.)?columbia\.edu
      - ^https://guides.gradescope\.com
      - ^https://stackoverflow\.com
      - ^https://(.+\.)?substack\.com
      # timeouts
      - ^https://data.cityofnewyork.us
      # encoding error...?
      - ^https://www.nyc.gov/assets/planning/download/pdf/data-maps/open-data/pluto_datadictionary.pdf
    linkcheck_report_timeouts_as_broken: false
    # https://myst-parser.readthedocs.io/en/latest/syntax/cross-referencing.html#implicit-targets
    myst_heading_anchors: 4
    # https://jupyterbook.org/en/stable/advanced/html.html#manually-specify-extra-files-folders-to-be-included-in-a-website
    html_extra_path:
      - assets
    suppress_warnings:
      # linkcheck doesn't like plotly HTML outputs
      - mystnb.mime_priority
      # Plotly gives a custom MIME Type
      # https://plotly.com/python/renderers/#plotly_mimetype
      - mystnb.unknown_mime_type
